src/etc/inc/pfsense-utils.inc

```diff
@@ -3562,4 +3562,13 @@ function send_user_download($type = 'data', $content, $userfilename = "", $conte
 	exit;
 }
 
+// Test whether the hostname in a URL can be resolved with a very short timeout
+function is_url_hostname_resolvable($url) {
+	$urlhostname = parse_url($url, PHP_URL_HOST);
```

Comment:
I would like to see one more test after this to ensure the result from `parse_url()` is sane. It can return `false` if there is a problem. Something as easy as this would suffice, on the line under this one:

    	if (empty($urlhostname)) {
    		return false;
    	}

That way it won't try to run a potentially invalid value through `gethostbyname()`.